internal/proxy/ui_requests.go

```diff
@@ -224,6 +271,27 @@ func (p *Proxy) loginJenkins(pci CacheItem, osoToken string, requestLogEntry *lo
 	return resp.StatusCode, resp.Cookies(), err
 }
 
+func clearCookiesMatching(w http.ResponseWriter, r *http.Request, expire func(cookie *http.Cookie) bool) {
+	for _, cookie := range r.Cookies() {
+		if expire(cookie) {
+			cookie.Expires = time.Unix(0, 0)
+		}
+		http.SetCookie(w, cookie)
+	}
+}
+
+func isSessionCookie(c *http.Cookie) bool {
+	return strings.HasPrefix(c.Name, SessionCookie)
+}
+
+func isIdledCookie(c *http.Cookie) bool {
+	return c.Name == CookieJenkinsIdled
+}
+
+func anyCookie(_ *http.Cookie) bool {
+	return true
+}
```

Comment:
It seems to me that current method will clear all cookies set by jenkins.openshift.io, is that right?

Reply:
all cookies will be cleared **iff** clearCookiesMatching is passed anyCookie. I am thinking of doing this when `token_json` is passed in the query and jenkins is idled .. I will revisit this thought since I think there could be jenkins-preference cookie that may still be valid.